src/main/java/kitchenpos/application/TableGroupService.java

```diff
@@ -1,90 +1,48 @@
 import org.springframework.transaction.annotation.Transactional;
-import org.springframework.util.CollectionUtils;
 
-import kitchenpos.dao.OrderDao;
-import kitchenpos.dao.OrderTableDao;
-import kitchenpos.dao.TableGroupDao;
-import kitchenpos.domain.OrderStatus;
 import kitchenpos.domain.OrderTable;
 import kitchenpos.domain.TableGroup;
+import kitchenpos.dto.TableGroupRequest;
+import kitchenpos.dto.TableGroupResponse;
+import kitchenpos.repository.OrderTableRepository;
+import kitchenpos.repository.TableGroupRepository;
 
 @Service
+@Transactional(readOnly = true)
 public class TableGroupService {
-	private final OrderDao orderDao;
-	private final OrderTableDao orderTableDao;
-	private final TableGroupDao tableGroupDao;
+	private final OrderTableRepository orderTableRepository;
+	private final TableGroupRepository tableGroupRepository;
 
-	public TableGroupService(final OrderDao orderDao, final OrderTableDao orderTableDao,
-		final TableGroupDao tableGroupDao) {
-		this.orderDao = orderDao;
-		this.orderTableDao = orderTableDao;
-		this.tableGroupDao = tableGroupDao;
+	public TableGroupService(
+		OrderTableRepository orderTableRepository, TableGroupRepository tableGroupRepository) {
+		this.orderTableRepository = orderTableRepository;
+		this.tableGroupRepository = tableGroupRepository;
 	}
 
 	@Transactional
-	public TableGroup create(final TableGroup tableGroup) {
-		final List<OrderTable> orderTables = tableGroup.getOrderTables();
-
-		if (CollectionUtils.isEmpty(orderTables) || orderTables.size() < 2) {
-			throw new IllegalArgumentException("주문 테이블이이 둘 이상이어야 단체지정을 할 수 있습니다.");
-		}
-
-		final List<Long> orderTableIds = orderTables.stream()
-			.map(OrderTable::getId)
-			.collect(Collectors.toList());
-
-		final List<OrderTable> savedOrderTables = orderTableDao.findAllByIdIn(orderTableIds);
-
-		if (orderTables.size() != savedOrderTables.size()) {
-			throw new IllegalArgumentException("주문 테이블과 저장된 주문 테이블의 갯수가 다릅니다.");
-		}
-
-		for (final OrderTable savedOrderTable : savedOrderTables) {
-			if (!savedOrderTable.isEmpty() || Objects.nonNull(savedOrderTable.getTableGroupId())) {
-				throw new IllegalArgumentException("주문테이블이 단체지정이 되어있거나, 비어있지 않은 테이블입니다.");
-			}
-		}
-
-		tableGroup.setCreatedDate(LocalDateTime.now());
-
-		final TableGroup savedTableGroup = tableGroupDao.save(tableGroup);
-
-		final Long tableGroupId = savedTableGroup.getId();
-		for (final OrderTable savedOrderTable : savedOrderTables) {
-			savedOrderTable.setTableGroupId(tableGroupId);
-			savedOrderTable.setEmpty(false);
-			orderTableDao.save(savedOrderTable);
-		}
-		savedTableGroup.setOrderTables(savedOrderTables);
+	public TableGroupResponse create(final TableGroupRequest tableGroupRequest) {
+		List<Long> orderTableIds = tableGroupRequest.getOrderTableIds();
+		List<OrderTable> orderTables = orderTableIds.stream().map(this::findOrderTable).collect(Collectors.toList());
+		TableGroup tableGroup = new TableGroup(LocalDateTime.now(), orderTables);
```

Comment:
```suggestion
		TableGroup tableGroup = new TableGroup(orderTables);
```
클라이언트가 생성시각을 전달하는게 아니라 객체 스스로 생성시각을 정하면 어떨까요

Reply:
다음 단계에서 반영하도록하겠습니다 😄